components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -632,6 +632,21 @@ public void addResidentIdP(IdentityProvider identityProvider, String tenantDomai
     public void updateResidentIdP(IdentityProvider identityProvider, String tenantDomain)
             throws IdentityProviderManagementException {
 
+        for (IdentityProviderProperty idpProp : identityProvider.getIdpProperties()) {
+            if (StringUtils.equals(idpProp.getName(), IdentityApplicationConstants.SESSION_IDLE_TIME_OUT)) {
+                if (StringUtils.isBlank(idpProp.getValue()) || !StringUtils.isNumeric(idpProp.getValue()) ||
+                        StringUtils.equals(idpProp.getValue().trim(), "0")) {
```

Comment:
Haven't checked for < 0. What you can do is use Integer.parseInt and check > 0. Since you've already verified isNumeric you can't get an exception.